### PR TITLE
feat: make file utils

### PR DESCRIPTION
**Pull request by WiemKhlifi**
## What?
Add new functions to create environments using the `create_envs_utils` file in ` mava.utils`.
## Why?
Instead of creating different file systems for each environment, we use a single file system and remove the common lines that create the environments. 
## How?
* Added a `create_envs_utils` file in ` mava.utils`: Create environments from config files. 
* Remove the common lines from systems files and use the `create_environment ` function.
* Edit configs to differentiate between the **ippo** and **mappo** systems.

### CODE REVIEW

File: mava/utils/create_envs_utils.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    eval_env = jumanji.make(env_name, generator=generator)
+    eval_env = wrapper(eval_env)
+
+    # Add agent id to observation.
+    if config["system"]["add_agent_id"]:
+        env = AgentIDWrapper(env)
+        eval_env = AgentIDWrapper(eval_env)
+
+    # Somehow it didn't work, only after switching the two ifs or do return GlobalStateWrapper(env)
+    if config["system"]["system_name"] in ["ff_mappo", "rec_mappo"]:
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:63]: I think system name isn't the way to do this, rather `add_global_state` so we don't have to check each system and then its also consistent with `add_agent_id`

File: mava/utils/create_envs_utils.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+from mava.wrappers.shared import AgentIDWrapper, GlobalStateWrapper, LogWrapper
+
+# Registry mapping environment names to their generator and wrapper classes.
+JumanjiRegistry = {
+    "RobotWarehouse-v0": {"generator": RwareRandomGenerator, "wrapper": RwareWrapper},
+    "LevelBasedForaging-v0": {"generator": LbfRandomGenerator, "wrapper": LbfWrapper},
+}
+
+
+def create_jumanji_env(env_name: str, config: Dict) -> Tuple[Environment, Environment]:
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:37]:
```suggestion
def make_jumanji_env(env_name: str, config: Dict) -> Tuple[Environment, Environment]:
```

File: mava/utils/create_envs_utils.py
Comment (sash-a) [inline mava/utils/create_envs_utils.py:1]: rename to `make_env.py`

File: mava/utils/create_envs_utils.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    Create a Jumanji environments for training and evaluation.
+    Args:
+        env_name (str): The name of the environment to create.
+        config (Dict): The configuration of the environment.
+    Returns:
+        A tuple of the environments.
+    """
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:45]: nit:
```suggestion
    """
    Create a Jumanji environments for training and evaluation.
    
    Args:
        env_name (str): The name of the environment to create.
        config (Dict): The configuration of the environment.

    Returns:
        A tuple of the environments.
    """
```

File: mava/utils/create_envs_utils.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    env_name = config["env"]["env_name"]
+    if env_name in JumanjiRegistry:
+        return create_jumanji_env(env_name, config)
+    else:
+        raise ValueError(f"{env_name} is not a supported environment.")
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:98]: Very nice :fire:

File: mava/utils/create_envs_utils.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+from jumanji.wrappers import AutoResetWrapper
+
+from mava.wrappers.jumanji import LbfWrapper, RwareWrapper
+from mava.wrappers.shared import AgentIDWrapper, GlobalStateWrapper, LogWrapper
+
+# Registry mapping environment names to their generator and wrapper classes.
+JumanjiRegistry = {
+    "RobotWarehouse-v0": {"generator": RwareRandomGenerator, "wrapper": RwareWrapper},
+    "LevelBasedForaging-v0": {"generator": LbfRandomGenerator, "wrapper": LbfWrapper},
+}
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:34]: Also very nice :fire:

File: mava/utils/create_envs_utils.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        env = GlobalStateWrapper(env)
+        eval_env = GlobalStateWrapper(env)
+
+    env = AutoResetWrapper(env)
+    env = LogWrapper(env)
+
+    return env, eval_env
+
+
+def create_jaxmarl_env(config: Dict) -> Environment:
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:73]:
```suggestion
def make_jaxmarl_env(config: Dict) -> Environment:
```

File: mava/utils/create_envs_utils.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    Create a JAXMARL environment.
+    Args:
+        config (Dict): The configuration of the environment.
+    Returns:
+        A JAXMARL environment.
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:79]:
```suggestion
    Create a JAXMARL environment.
    
    Args:
        config (Dict): The configuration of the environment.
        
    Returns:
        A JAXMARL environment.
```

File: mava/utils/create_envs_utils.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    # Placeholder for creating JAXMARL environment.
+    pass
+
+
+def create_environment(config: Dict) -> Tuple[Environment, Environment]:
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:85]:
```suggestion
def make(config: Dict) -> Tuple[Environment, Environment]:
```

File: mava/utils/create_envs_utils.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+from jumanji.environments.routing.robot_warehouse.generator import (
+    RandomGenerator as RwareRandomGenerator,
+)
+from jumanji.wrappers import AutoResetWrapper
+
+from mava.wrappers.jumanji import LbfWrapper, RwareWrapper
+from mava.wrappers.shared import AgentIDWrapper, GlobalStateWrapper, LogWrapper
+
+# Registry mapping environment names to their generator and wrapper classes.
+JumanjiRegistry = {
```
Comment (sash-a) [inline mava/utils/create_envs_utils.py:31]:
```suggestion
_jumanji_registry = {
```

**Review by sash-a — CHANGES_REQUESTED** [review]
Very nice, just minor things like `make...` instead of `create...` to stay consistent with gym, but this PR is almost ready to go :fire:

**Review by RuanJohn — COMMENTED** [review]
Very very nice @WiemKhlifi :fire: 
I agree with Sasha's suggestion. Also, feel free to change the file names.

**Comment by WiemKhlifi** [comment]
> Very nice, just minor things like `make...` instead of `create...` to stay consistent with gym, but this PR is almost ready to go 🔥

All the changes have been made 🙌 . I think this report is now ready or not yet?

**Review by sash-a — APPROVED** [review]
Looks great, glad we have a really nice way of switching between envs now :fire:

**Review by RuanJohn — APPROVED** [review]
Very nice! Thank you @WiemKhlifi :fire:

**Review by RuanJohn — APPROVED** [review]

**Pull request merged**